### DISCUSSION TITLE
68. 데이터베이스에서 ‘키’란 무엇인가요?

### DISCUSSION THREAD

**Discussion by cutehumanS2** [jewelry-box/cs-interview#70] [General]
- 키의 종류에는 무엇이 있나요?
- 기본 키, 후보 키, 대체 키, 슈퍼 키를 설명해 주세요.

**Comment by eun-byeol** [comment]
데이터베이스에서 키는 `행을 구분하는 유일한 식별자`를 의미합니다.
대표적으로 `슈퍼키`, `후보키`, `기본키`, `대체키`, `외래키`가 있습니다.
슈퍼키는 `유일성`을 만족하는 집합입니다.
후보키는 `유일성`*과 `최소성`*을 만족하는 집합입니다. 슈퍼키 중 최소성을 만족하는 것이 후보키가 됩니다.
기본키는 후보키 중 기본적으로 사용할 키입니다.
대체키는 후보키 중 기본키로 선택되지 못한 키입니다.
외래키는 다른 릴레이션의 기본키를 참조하는 속성의 집합입니다.

**Comment by eun-byeol** [comment]
![image](https://github.com/user-attachments/assets/fc7a0e90-5ceb-4c10-893e-1e129234ca42)
> 유일성
> 

각 행을 유일하게 식별할 수 있는 것

> 최소성
> 

키를 구성하는 속성 중 꼭 필요한 최소한의 속성들로만 키 구성하는 것
